Allow ignored extensions to be defined by user

estebanz01 opened the pull request:
👋 Hello again!

Just another PR to complement #2. We want to specify which file extensions should be ignored when uploading to S3, so this is just a commit for that!

Waiting for suggestions and comments! 🍺

tlatsas commented on lib/webpacker_uploader/instance.rb:
```diff
@@ -3,26 +3,37 @@
-  # TODO make this configurable
-  IGNORE_EXTENSION = %w[.map].freeze
-  public_constant :IGNORE_EXTENSION
+  DEFAULT_IGNORE_EXTENSION = %w[.map].freeze
```
I would completely remove the `IGNORE_EXTENSION` and `DEFAULT_IGNORE_EXTENSION` constants. We can directly use the `ignored_extensions` class attribute below.

tlatsas commented on lib/webpacker_uploader/instance.rb:
```diff
@@ -3,26 +3,37 @@
+                           else
+                             ignored_extensions
+                           end
+
```
I think we can simplify and skip all these checks. We can directly use the `ignored_extensions`. This will have `.map` ignored by default as is correctly set in line 9.

I will also document how to configure this, so I don't think the array check is important.

I would like to have this set to an empty array in the future, to allow for map file uploading, but the content type logic does not support this right now.

estebanz01 replied:
Right now, `Rack::Mime` returns `application/octet-stream` for source map files. I think that type makes sense for this type of files (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#applicationoctet-stream).
```ruby
.7.2 :011 > Rack::Mime.mime_type('.js.map')
 => "application/octet-stream" 
```

tlatsas commented on lib/webpacker_uploader/instance.rb:
```diff
@@ -3,26 +3,37 @@
         logger.info("Processing: #{file_path}")
-        provider.upload!(path, file_path, content_type_for(path)) unless name.end_with?(*IGNORE_EXTENSION)
+
+        unless name.end_with?(*extensions_to_ignore)
+          provider.upload!(path, file_path, content_type_for(path))
+        end
```
We can remove this if I think, it is covered by the outer clause.

Something like this will suffice:

```ruby
      if name.end_with?(*ignored_extensions)
        logger.info("Skipping: #{file_path}")
      else
        logger.info("Processing: #{file_path}")

        provider.upload!(path, file_path, content_type_for(path))
      end
```

estebanz01 commented:
@tlatsas linting issues fixed.

tlatsas reviewed (changes requested):
Thank you @estebanz01 for this, this is something I was planning to implement!
I think we can simplify it a bit. Check my comments.

estebanz01 commented:
@tlatsas in my previous PR (#2 ) I introduced a bug where the path used to upload the file is `nil` if `prefix` is also nil. I fixed that one here. This one ready as well 👌

tlatsas commented:
🎉  🎉
